### PR TITLE
V1.3: Find-by-ingredient

### LINKED ISSUE #145: V1.3: Find by ingredient

### PR DESCRIPTION
**Pull request by filbertphang**
Closes #145 

- [x] Implement find by ingredient
- [x] Add tests
- [x] Fix checkstyle
- [ ] Update DG/UG (to be done in separate branch)

### CODE REVIEW

**Comment by filbertphang** [comment]
(squashed merge commit from master to v1.3 find to prevent this PR from having 130+ commits)

**Comment by SeeuSim** [comment]
you might want to rebase your upstream branch to the latest `master`, so that you don't overwrite changes

**Review by ZHTang29 — APPROVED** [review]
Good work, LGTM!

File: src/main/java/seedu/recipe/logic/parser/FindCommandParser.java
Comment (alson001) [inline src/main/java/seedu/recipe/logic/parser/FindCommandParser.java:1]: LGTM

File: src/main/java/seedu/recipe/model/recipe/ingredient/Ingredient.java [hunk 1]
```diff
@@ -1,9 +1,98 @@
...
+    public void setCommonName(String s) {
+        assert s != null;
+        checkArgument(isValidIngredientName(s), MESSAGE_CONSTRAINTS);
+        this.commonName = s;
+    }
+
+    public String getNames() {
+        String s = String.format("%s %s", this.name, this.commonName);
+        return s;
+    }
```
Comment (alson001) [inline src/main/java/seedu/recipe/model/recipe/ingredient/Ingredient.java:97]: Seems like it can work.

**Review by alson001 — APPROVED** [review]
LGTM

**Pull request merged**